files/image_config/config-setup/config-setup

```diff
@@ -0,0 +1,399 @@
+# by SONiC SWSS. It also performs configuration                           #
+# migration.                                                              #
+#                                                                         #
+###########################################################################
+
+# Initialize constants
+CONFIG_DB_INDEX=4
+UPDATEGRAPH_CONF=/etc/sonic/updategraph.conf
+CONFIG_DB_JSON=/etc/sonic/config_db.json
+MINGRAPH_FILE=/etc/sonic/minigraph.xml
+TMP_ZTP_CONFIG_DB_JSON=/tmp/ztp_config_db.json
+FACTORY_DEFAULT_HOOKS=/etc/config-setup/factory-default-hooks.d
+CONFIG_PRE_MIGRATION_HOOKS=/etc/config-setup/config-migration-pre-hooks.d
+CONFIG_POST_MIGRATION_HOOKS=/etc/config-setup/config-migration-post-hooks.d
+CONFIG_SETUP_VAR_DIR=/var/lib/config-setup
+CONFIG_SETUP_PRE_MIGRATION_FLAG=${CONFIG_SETUP_VAR_DIR}/pending_pre_migration
+CONFIG_SETUP_POST_MIGRATION_FLAG=${CONFIG_SETUP_VAR_DIR}/pending_post_migration
+CONFIG_SETUP_INITIALIZATION_FLAG=${CONFIG_SETUP_VAR_DIR}/pending_initialization
+
+# Command usage and help
+usage()
+{
+    cat << EOF
+ Usage:  config-setup < backup | boot | factory >
+
+         backup  - Take a backup copy of SONiC configuration.
+         boot    - Initialize/migrate SONiC configuration during system boot.
+         factory - Create factory default SONiC configuration and save it to
+                   to ${CONFIG_DB_JSON}.
+EOF
+}
+
+# run given script
+run_hook() {
+    local script="$1"
+    local exit_status=0
+
+    if [ -f $script ]; then
+        # Check hook for syntactical correctness before executing it
+        /bin/bash -n $script
+        exit_status=$?
+        if [ "$exit_status" -eq 0 ]; then
+            . $script
+        fi
+        exit_status=$?
+    fi
+
+    if [ -n "$exit_status" ] && [ "$exit_status" -ne 0 ]; then
+        echo "$script returned non-zero exit status $exit_status"
+    fi
+
+    return $exit_status
+}
+
+# run scripts in given directory
+run_hookdir() {
+    local dir="$1"
+    local progress_file="$2"
+    local exit_status=0
+
+    if [ -d "$dir" ]; then
+        if [ -n $progress_file ]; then
+            [ ! -d  $(dirname $progress_file) ] && mkdir -p $(dirname $progress_file)
+            [ ! -e $progress_file  ] &&  run-parts --list $dir > $progress_file
+            SCRIPT_LIST=$(cat $progress_file)
+        else
+            SCRIPT_LIST=$(run-parts --list $dir)
+        fi
+
+        for script in $SCRIPT_LIST; do
+            run_hook $script
+            exit_status=$((exit_status|$?))
+            script_name=$(basename $script)
+            sed -i "/$script_name/d" $progress_file
+        done
+        [ -n $progress_file ] && [ "$(cat ${progress_file})" = "" ] && rm -f ${progress_file}
+    fi
+
+    return $exit_status
+}
+
+# Reload minigraph.xml file on disk
+reload_minigraph()
+{
+    echo "Reloading minigraph..."
+    if [ ! -f /etc/sonic/init_cfg.json ]; then
```

Comment:
can't the command "config reload or config load -m <>" be used ?
This code here replicated steps from load_minigraph() function in sonic-utilities/config/main.py
If someone updates that part, they should also be updating it here. Doing it like this opens doorway to config load getting out of sync between "config load" and through ztp

Reply:
@rameshsanth

The scope of this change was to move the code from graphservice to config-setup service keeping the functionality as is for most part. Changing logic to use "config load" and  "config load_mingraph" can be done as part of a different PR.

Reply:
I agree with @rameshsanth , created an issue to merge load_minigraph in config-setup and update_minigraph and assigned to you. @rajendra-dendukuri